docs/destruction.md

```diff
@@ -0,0 +1,39 @@
+# Project Approach to Object Destruction
```

Comment:
This is great! Can we link to it from the development readme or somewhere?

Reply:
Hmm, I'm not ready to link to it yet. Not sure where it would be linked to from. It's sort of the first of its kind.